child_compassion/models/household.py

```diff
@@ -215,6 +216,9 @@ def process_commkit(self, commkit_data):
     ##########################################################################
     @api.model
     def create(self, vals):
+        if vals['role'] not in self._get_roles():
```

Comment:
`vals.get('role')` to avoid errors in case role is not present in vals.
`self._get_roles()` returns a list of tuples. You should get a list of only the first values of the tuple for the comparison to work!